supler-js/src/CreateFormFromJson.ts

```diff
@@ -5,26 +5,43 @@ module Supler {
     constructor(private renderOptionsGetter:RenderOptionsGetter,
                 private i18n:I18n,
                 private validatorFnFactories:any,
-                private fieldsOptions:FieldsOptions) {
+                private fieldsOptions:FieldsOptions,
+                private fieldOrder: string[][]) {
     }
 
     renderForm(meta,
                formJson,
                formElementDictionary:FormElementDictionary = new FormElementDictionary()):RenderFormResult {
-      var fields = formJson.fields;
-      var fieldCount = fields.length;
+      var fields = formJson.fields.slice();
 
-      var html = this.generateMeta(meta);
+      var rowsHtml = '';
 
-      for (var i = 0; i < fieldCount; i++) {
-        var fieldJson = fields[i];
-        var fieldResult = this.fieldFromJson(fieldJson, formElementDictionary, false);
-        if (fieldResult) {
-          html += fieldResult + '\n';
-        }
+      (this.fieldOrder || formJson.fieldOrder).forEach(row => {
+        rowsHtml += this.row((<string[]>row).map(fieldName => this.findField(fieldName, fields)),
+          formElementDictionary, this.renderOptionsGetter.defaultRenderOptions())
+      });
+
+      if (fields.filter(f => f).length > 0) {
```

Comment:
it will remove all undefined objects - if you have an array var arr = [a, b, c] and you do ``delete arr[0]`` it will become [undefined, b, c] - in order to get rid of undefined elements, you can filter by f => f (undefined evaluates to false in js-truth)